test/test_utils/security.py

```diff
@@ -101,12 +101,14 @@ def __init__(
     def __eq__(self, other):
         assert type(self) == type(other), f"Types {type(self)} and {type(other)} mismatch!!"
         ## Ignore version key in package_details as it might represent the version of the package existing in the image
-        ## and might differ from  image to image, even when the vulnerability is same.
+        ## and might differ from  image to image, even when the vulnerability is same. 
+        ## Also ignore the title key of the vulnerablitiy, because, sometimes, 1 vulnerability impacts multiple packages.
+        ## In that case, the title key is generated by ECR scans by mentioning the name of all packages in a random order. This fails during comparison.
         if test_utils.check_if_two_dictionaries_are_equal(
             dataclasses.asdict(self.package_details), dataclasses.asdict(other.package_details), ignore_keys=["version"]
         ):
             return test_utils.check_if_two_dictionaries_are_equal(
-                dataclasses.asdict(self), dataclasses.asdict(other), ignore_keys=["package_details"]
+                dataclasses.asdict(self), dataclasses.asdict(other), ignore_keys=["package_details", "title"]
```

Comment:
Let us implement an approach that retrieves the unique key identifier(s) of vulnerabilities instead of using ignore list that may continue to grow.